hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieTimeline.java

```diff
@@ -133,6 +137,21 @@
    */
   HoodieTimeline getCommitsAndCompactionTimeline();
 
+  /**
+   * Timeline to just include commits (commit/deltacommit), replace and compaction actions.
+   *
+   * @return
+   */
+  HoodieDefaultTimeline getWriteActionTimeline();
```

Comment:
see earlier comment on whether we need this method.

Reply:
To expand, I am wondering if we should just include replacecommit within `getCommitsAndCompactionTimeline()`. Most of its callers are around compaction/savepoint/restore etc. So we may not be seeing some cases here. 

Things work for now, since filterCompletedInstants() etc are including replace commit in the timeline when filtering for queries. Semantically, if replace is a commit level action that can add new data to the timeline, then we should just treat it like delta commit IMO. 

Would anything break if we did do that?

Reply:
I removed and included getCommitsAndCompactionTimeline. I think we will run into some edge cases for MOR tables where something would break. But don't have concrete examples. We can run it for a while on MOR table and see if works.